app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationDataTransformer.kt

```diff
@@ -0,0 +1,353 @@
+            generatedDate = Dates.formatAsISO8601_Date(Date()),
+            canvasserName = partnerInformation.canvasserName,
+            voterRegistration = buildVoterRegistration()
+        )
+    }
+
+    private fun buildVoterRegistration(): VoterRegistration {
+        val mailingAddress = if (addressData.isMailingAddressDifferent) {
+            addressData.mailingAddress ?: throw InvalidRegistrationException(
+                "addressData.isMailingAddressDifferent is null",
+                R.string.different_mailing_address_not_filled_out
+            )
+        } else {
+            null
+        }
+
+        // TODO Is this the way that we should be specifying race? It's required in the API, but not in the app
+        val race = (additionalInfoData.race ?: Race.DECLINE).toString()
+
+        val party = when (additionalInfoData.party) {
+            Party.OTHER_PARTY -> additionalInfoData.otherPoliticalParty
+                ?: throw InvalidRegistrationException(
+                    "otherPoliticalParty is null",
+                    R.string.must_write_in_party
+                )
+            else -> additionalInfoData.party.toString()
+        }.toLowerCase() // TODO i
+
+        // TODO NO_WRAP was specified in ApiSignature, but probably should be in the API docs as well
+        val signature = Base64.encodeToString(reviewData.signature, Base64.NO_WRAP)
+
+        return VoterRegistration(
+            registrationHelper = buildRegistrationHelper(),
+            dateOfBirth = Dates.formatAsISO8601_ShortDate(newRegistrationData.birthday), //TODO DateAdapter used short date, is this correct?
+            mailingAddress = mailingAddress?.toApiAddressData(),
+            previousRegistrationAddress = addressData.previousAddress?.toApiAddressData(),
+            registrationAddress = addressData.homeAddress.toApiAddressData(),
+            registrationAddressIsMailingAddress = !addressData.isMailingAddressDifferent,
+            name = newRegistrationData.name.toApiName(),
+            previousName = newRegistrationData.previousName?.toApiName(),
+            gender = Gender.fromPrefix(newRegistrationData.name.title).toString(),
+            race = race, // TODO race is required in the API, but not in the app?
+            party = party,
+            voterClassifications = buildVoterClassifications(),
+            signature = Signature(
+                mimeType = "image/png",
+                image = signature
+            ),
+            voterIds = buildVoterIds(),
+            contactMethods = buildContactMethods(),
+            additionalInfo = buildAdditionalInfo()
+        )
+
+    }
+
+    private fun buildRegistrationHelper(): RegistrationHelper? {
+        if (!assistanceData.hasSomeoneAssisted) return null // TODO is this correct, or should this be based on the canvasser and always non-null?
```

Comment:
this is correct, it's not based on the canvassed it's based on the checkbox in one of the registration pages